libenkf/src/enkf_obs.c

```diff
@@ -582,140 +582,194 @@ static void enkf_obs_update_keys( enkf_obs_type * enkf_obs ) {
 
-    conf_instance_free(enkf_conf      );
-    conf_class_free(   enkf_conf_class);
+  for(int i = 0; i < num_block_obs; i++) {
+    const char * obs_key = stringlist_iget(block_obs_keys, i);
+    const conf_instance_type * block_obs_conf = conf_instance_get_sub_instance_ref(enkf_conf, obs_key);
+    obs_vector_type * obs_vector = obs_vector_alloc_from_BLOCK_OBSERVATION(block_obs_conf,
+                                                                           enkf_obs->grid,
+                                                                           enkf_obs->obs_time,
+                                                                           enkf_obs->refcase,
+                                                                           enkf_obs->ensemble_config);
+    if (obs_vector != NULL)
+      enkf_obs_add_obs_vector(enkf_obs, obs_vector);
+  }
+  stringlist_free(block_obs_keys);
+}
 
-    enkf_obs_update_keys( enkf_obs );
-    return true;
-  } else
-    return false;
+
+
+/** Handle GENERAL_OBSERVATION instances. */
+static void handle_general_observation(enkf_obs_type * enkf_obs,
+                                       conf_instance_type * enkf_conf) {
+  stringlist_type * block_obs_keys = conf_instance_alloc_list_of_sub_instances_of_class_by_name(enkf_conf,
+                                                                                                "GENERAL_OBSERVATION");
+  int num_block_obs = stringlist_get_size(block_obs_keys);
+
+  for(int i = 0; i < num_block_obs; i++) {
+    const char * obs_key = stringlist_iget(block_obs_keys, i);
+    const conf_instance_type * gen_obs_conf = conf_instance_get_sub_instance_ref(enkf_conf,
+                                                                                 obs_key);
+
+    obs_vector_type * obs_vector = obs_vector_alloc_from_GENERAL_OBSERVATION(gen_obs_conf,
+                                                                             enkf_obs->obs_time,
+                                                                             enkf_obs->ensemble_config);
+    if (obs_vector != NULL)
+      enkf_obs_add_obs_vector(enkf_obs, obs_vector);
+  }
+  stringlist_free(block_obs_keys);
 }
 
 
 
 
+/**
+   This function will load an observation configuration from the
+   observation file @config_file.
+
+   If called several times during one invocation the function will
+   start by clearing the current content.
+*/
+bool enkf_obs_load(enkf_obs_type * enkf_obs ,
+                   const char * config_file,
+                   double std_cutoff) {
+
+  if (!enkf_obs->valid)
+    return false;
+
+  int last_report                      = enkf_obs_get_last_restart( enkf_obs );
+  conf_class_type    * enkf_conf_class = enkf_obs_get_obs_conf_class();
+  conf_instance_type * enkf_conf       = conf_instance_alloc_from_file(enkf_conf_class, "enkf_conf", config_file);
+
+  if(!conf_instance_validate(enkf_conf))
+    util_abort("Can not proceed with this configuration.\n");
```

Comment:
These `util_abort( )` calls should all include a `%s` which resolvs to `__func__`:
```C
util_abort("%s: Can not proceed with this configuration.\n", __func__);
```

Reply:
Yes, I actually thought so ... I will update them whenever I find them.  Please stay put for another commit. :)